### PR TITLE
Adds Collapsible AM rifle. Createes new uplink entry

### PR DESCRIPTION
**Pull request by LorenLuke**
Adds Collapsible AM rifle. Adds case with parts and 5 shots to Uplink. Separates it and intended merc version, both costing default telecrystal amount. ~~Does NOT make either rifle antagonist-type specific.~~ Does IN FACT make either rifle antagonist-type specific.

### CODE REVIEW

File: code/modules/projectiles/guns/projectile.dm [hunk 1]
```diff
@@ -32,13 +32,14 @@
...
-		for(var/i in 1 to max_shells)
-			loaded += new ammo_type(src)
-	if(ispath(magazine_type) && (load_method & MAGAZINE))
-		ammo_magazine = new magazine_type(src)
+	if(starts_loaded)
+		if(ispath(ammo_type) && (load_method & (SINGLE_CASING|SPEEDLOADER)))
+			for(var/i in 1 to max_shells)
+				loaded += new ammo_type(src)
+		if(ispath(magazine_type) && (load_method & MAGAZINE))
+			ammo_magazine = new magazine_type(src)
```
Comment (Anewbe) [inline code/modules/projectiles/guns/projectile.dm]: Why exactly did you change this?

File: code/game/objects/items/weapons/storage/uplink_kits.dm [hunk 2]
```diff
@@ -281,4 +281,17 @@
...
+	name = "secure briefcase"
+
+/obj/item/weapon/storage/secure/briefcase/rifle/New()
+	..()
+	new /obj/item/sniper_rifle_part/barrel(src)
+	new /obj/item/sniper_rifle_part/stock(src)
+	new /obj/item/sniper_rifle_part/trigger_group(src)
+
+	for(var/i = 1 to 5)
+		new /obj/item/ammo_casing/a145(src)
```
Comment (Anewbe) [inline code/game/objects/items/weapons/storage/uplink_kits.dm]: This is going to spawn a massive briefcase, maybe make a single item to hold the ammo_casings.

File: code/modules/projectiles/guns/projectile/sniper/collapsible_sniper.dm [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+	var/obj/item/sniper_rifle_part/stock = null
+	var/obj/item/sniper_rifle_part/trigger_group = null
+	var/part_count = 1
+
+
+/obj/item/sniper_rifle_part/barrel
+	name = "AM rifle barrel"
+	icon_state = "heavysniper-barrel"
+
+	New()
```
Comment (Anewbe) [inline code/modules/projectiles/guns/projectile/sniper/collapsible_sniper.dm]: Absolute pathing.

File: code/modules/projectiles/guns/projectile/sniper/collapsible_sniper.dm [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+		..()
+		barrel = src
+
+/obj/item/sniper_rifle_part/stock
+	name = "AM rifle stock"
+	icon_state = "heavysniper-stock"
+
+	New()
```
Comment (Anewbe) [inline code/modules/projectiles/guns/projectile/sniper/collapsible_sniper.dm]: Absolute pathing.

File: code/modules/projectiles/guns/projectile/sniper/collapsible_sniper.dm [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+		..()
+		stock = src
+
+/obj/item/sniper_rifle_part/trigger_group
+	name = "AM rifle trigger assembly"
+	icon_state = "heavysniper-trig"
+
+	New()
```
Comment (Anewbe) [inline code/modules/projectiles/guns/projectile/sniper/collapsible_sniper.dm]: Absolute pathing.

File: code/modules/projectiles/guns/projectile.dm (hunk 1 above)
Comment (LorenLuke) [inline code/modules/projectiles/guns/projectile.dm]: Fun fact, if I didn't, you could disassemble and reassemble the rifle and get a free bullet every time.

File: code/game/objects/items/weapons/storage/uplink_kits.dm (hunk 2 above)
Comment (LorenLuke) [inline code/game/objects/items/weapons/storage/uplink_kits.dm]: It actually ends up being one ammo-casing (small item) larger in size difference from a regular briefcase, to where if I did 1 to 4, it'd be identical in size.

Comment (Anewbe) [inline code/game/objects/items/weapons/storage/uplink_kits.dm]: Do that, then.

File: code/modules/projectiles/guns/projectile.dm (hunk 1 above)
Comment (Anewbe) [inline code/modules/projectiles/guns/projectile.dm]: Did you make sure the other weapons still act the same way as before?

Comment (LorenLuke) [inline code/modules/projectiles/guns/projectile.dm]: Yeah the 'starts loaded' is true by default, so even if called without an argument, they'd be loaded. So, this is the only instance that such would not be the case.

**Comment by LorenLuke** [comment]
@Anewbe 
>Testing

Repeatedly. 

Primary issue I couldn't seem to solve has to do with the uplink showing both the traitor-intended rifle and merc-intended rifle on the uplink, but the traitor intended one has a description for the uplink. I imagine people would figure it out.

**Comment by Anewbe** [comment]
The issue might be in here?
https://github.com/PolarisSS13/Polaris/blob/0bb15b27324a371804f8f625708cbe8bb03b00e0/code/datums/uplink/uplink_items.dm#L39

**Comment by Anewbe** [comment]
Is this ready to go?

**Comment by LorenLuke** [comment]
@Anewbe 
>Ready
Yes.

**Pull request merged**